modules/programs/ssh.nix

```diff
@@ -158,7 +166,9 @@ let
   });
 
   matchBlockStr = cf: concatStringsSep "\n" (
-    ["Host ${cf.host}"]
+    let
+      matchType = if cf.matchType != null then "Match ${cf.matchType} ${cf.host}" else "Host ${cf.host}";
```

Comment:
Do you know the difference between `Host abc` and `Match host abc` (or `Match originalhost abc` (or `Match canonical abc`))? It would be sweet if one of these were the same as `Host abc` so that the `null` case could be avoided.